### DISCUSSION TITLE
[Contract Verification] Verified contract on explorer.zksync impossible to verify on era.zksync.network

### DISCUSSION THREAD

**Discussion by L0GYKAL** [zkSync-Community-Hub/zksync-developers#851] [Contract Verification] [answered]
### Team or Project

Zyfi

### Environment

Mainnet

### Block Explorer

https://era.zksync.network/

### Issue Type

- [ ] Plugin
- [ ] Block Explorer

### Contract Address

0x9f4D380c867EBaed8140C332c78BF32Eb52A01Fb

### Compiler Type

Multi-part contract

### zkSolc Version

1.4.1

### Solc Version

0.8.2

### Contract Name

ZFIToken

### Contract Code

The contract is verified on "explorer.zksync.io":
https://explorer.zksync.io/address/0x9f4d380c867ebaed8140c332c78bf32eb52a01fb#contract
And the code is fully available on our github repository:
https://github.com/ondefy/Zyfi-Token-Zkforge (ZFIToken.sol and all its dependencies)
(The flag: bytecode_hash = "none" has to be used here)

### Constructor Arguments

None as per the deployment transaction:
https://era.zksync.network/tx/0x80571d7a0f39717cd56eb4df9962454e7694f9b2c0648d19e3e1416e4f592186

### Repo Link (Optional)

https://github.com/ondefy/Zyfi-Token-Zkforge

### Additional Details

The contract is already deployed and verified on "explorer.zksync.io" at this address:
https://explorer.zksync.io/address/0x9f4d380c867ebaed8140c332c78bf32eb52a01fb#contract
The flag: bytecode_hash = "none" has to be used here.

**Comment by alhenaai** [comment]
It appears that you are encountering issues verifying your contract on the ZKsync Era network at `https://era.zksync.network/` even though it is already verified on `https://explorer.zksync.io/`. This could be due to differences in how contract verification is handled across different ZKsync environments.

Here are a few steps you can take to troubleshoot and potentially resolve the verification issue on the ZKsync Era network:

1. **Ensure Compatibility**: Make sure that the contract code, including all dependencies, is compatible with the ZKsync Era environment. This includes checking the compiler versions and settings used.

2. **Compiler Settings**: Since you mentioned using the flag `bytecode_hash = "none"`, ensure that this setting is correctly applied in the ZKsync Era verification process. This setting can affect how the bytecode is compared during verification.

3. **Verify Contract Code and Address**: Double-check that the contract address and code on GitHub match exactly with what is deployed. Any slight variation in the code or compilation settings can cause verification to fail.

4. **Use Correct Verification Tools**: Utilize the appropriate tools and plugins for ZKsync Era. If you are using Hardhat, ensure that your `hardhat.config.js` is set up correctly for ZKsync Era, including network configurations and plugin versions.

5. **Consult Documentation**: Review the [ZKsync Era documentation on contract verification](https://docs.zksync.io/zksync-era/tooling/hardhat/plugins/hardhat-zksync-verify) for any additional settings or steps that might be required specifically for the ZKsync Era network.

If after following these steps you still face issues, it might be helpful to engage with the ZKsync developer community for more specific guidance or potential issues that others have encountered and resolved. You can join the discussions on [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth support.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment, marked as answer]
@L0GYKAL I do not think its possible to verify contract with no bytecode hash on Etherscan (e.g. "The flag: bytecode_hash = "none" has to be used here"). 

See here: https://book.getfoundry.sh/forge/deploying#verifying-contracts-with-no-bytecode-hash

**Reply by L0GYKAL** [reply]
Thank you very much, I completely missed it🙏
